### ISSUE TITLE
Can't supply kubeconfig path using KUBE_CONFIG_PATH env variable

### ISSUE THREAD

**Issue by v01t** [fluxcd/terraform-provider-flux#440]
Hello,

I faced situation when I can't dynamically define path to kubeconfig in flux provider without using extra (bridge) variable:
```
provider "flux" {
  kubernetes = {
    config_path = var.kubeconfig
  }
  git = {
    url = "ssh://${var.github_ssh_user}@${var.github_url}/${var.github_org}/${var.github_repository}.git"
    branch = var.github_branch
    ssh = {
      username = var.github_ssh_user
      private_key = tls_private_key.flux.private_key_pem
    }
  }
}
```
according to [documentation](https://registry.terraform.io/providers/fluxcd/flux/latest/docs#config_path) config_path for kubernetes attribute could be supplied using KUBE_CONFIG_PATH env variable, but it doesn't work together with git attribute:
```
provider "flux" {
  kubernetes = {
  }
  git = {
    url = "ssh://${var.github_ssh_user}@${var.github_url}/${var.github_org}/${var.github_repository}.git"
    branch = var.github_branch
    ssh = {
      username = var.github_ssh_user
      private_key = tls_private_key.flux.private_key_pem
    }
  }
}
```
lead to following error:
```
│ Error: Kubernetes Client
│
│   with flux_bootstrap_git.main,
│   on flux.tf line 14, in resource "flux_bootstrap_git" "main":
│   14: resource "flux_bootstrap_git" "main" {
│
│ invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable
```
same for
```
  kubernetes = {
    config_path = ""
  }
```

How I could supply path to kubeconfig without bridging value like:
$TF_VAR_kubeconfig > var.kubeconfig > kubernetes.config_path
and make it work directly:
$KUBE_CONFIG_PATH > kubernetes.config_path
in my case?

Version: `fluxcd/flux v0.25.3`

Thanks

**Comment by phillebaba** [comment]
The provider does not attempt to fetch attribute values from environment variables. The reason for this is that there is currently no good support to implement this in the new provider SDK. The docs are incorrect, I will create a PR that fixes it, sorry for the confusion.

**Comment by viceice** [comment]
why not doing something like the `kubernetes` provider?

https://github.com/hashicorp/terraform-provider-kubernetes/blob/5bc27bc9b9e2831cb43d838bf9017f14130dca65/kubernetes/provider.go#L90-L96

**Comment by viceice** [comment]
```go

			"config_path": {
				Type:          schema.TypeString,
				Optional:      true,
				DefaultFunc:   schema.EnvDefaultFunc("KUBE_CONFIG_PATH", nil),
				Description:   "Path to the kube config file. Can be set with KUBE_CONFIG_PATH.",
				ConflictsWith: []string{"config_paths"},
			},
```

https://github.com/fluxcd/terraform-provider-flux/blob/9c8106ca62882ee7c4da12ee09f7de227cc30f28/internal/provider/provider.go#L147-L150

**Comment by networkhermit** [comment]
Not only does the [kubernetes](https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs#argument-reference) provider, the [helm](https://registry.terraform.io/providers/hashicorp/helm/latest/docs#argument-reference) provider also supports setting a set of common kubernetes arguments via environment variables:

```text
KUBE_CONFIG_PATH
KUBE_CLIENT_CERT_DATA
KUBE_CLIENT_KEY_DATA
KUBE_CLUSTER_CA_CERT_DATA
KUBE_HOST
...
```

I think it's better if flux could align with the popular kubernetes and helm provider, as that improves friction.

**Comment by swade1987** [comment]
@v01t @networkhermit 

Is this missing functionality blocking your ability to use the provider? 

The reason I ask is that this provider uses github.com/hashicorp/terraform-plugin-framework 

The kubernetes and helm providers use github.com/hashicorp/terraform-plugin-sdk which is seen as the old way. 

See snippet from the plugin-sdk README below:

>For new provider development it is recommended to investigate [terraform-plugin-framework](https://github.com/hashicorp/terraform-plugin-framework), which is a reimagined provider SDK that supports additional capabilities. Refer to the [Which SDK Should I Use?](https://terraform.io/docs/plugin/which-sdk.html) documentation for more information about differences between SDKs.

**Comment by swade1987** [comment]
@networkhermit
I am going to track this enhancement in https://github.com/fluxcd/terraform-provider-flux/issues/655 and aim to have it released either in the upcoming 1.3.0 release or a fast follow in a 1.3.1 release.

**Comment by networkhermit** [comment]
> @networkhermit I am going to track this enhancement in #655 and aim to have it released either in the upcoming 1.3.0 release or a fast follow in a 1.3.1 release.

@swade1987
Thanks. I want to remind you that I can't reproduce this issue in my testing k3s cluster in case you missed my edited comment.

I just downgraded my provider version to fluxcd/flux v0.25.3 as the author once used but got no vein trying to reproduce the error. It bugs me.

I'm using `OpenTofu v1.6.2` and the cluster is already bootstrapped (I can't destroy the cluster as it have some services on it). I checked the `terraform plan` and there is no error. Maybe this error would happen in the actual terraform apply phase but currently I don't have any brand new cluster so I can't help verify it.

If you would work on this enhancement, please don't forget to reproduce it first.

Thanks again for the attention.
